From 38b8e109d94bfa6378a2027747900cf1312dfae7 Mon Sep 17 00:00:00 2001
From: "haolong.zhang" <haolong.zhang@ck-telecom.com>
Date: Wed, 3 Dec 2014 12:23:42 +0800
Subject: [PATCH] [Feature] enable rtxxx usb dongle wifi step 1: enable kernel
 driver
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[id]:NONE
[Type]
   FEATURE
[Products]
	EYES_BASELINE
[Requirement]
   N/A

[Solution Description]
   N/A

[Other Info]
   N/A

 要提交的变更：
	修改:         mediatek/config/mt6595/autoconfig/kconfig/platform

Change-Id: I0aefe9ca10f2478dfe4fd0bdc13926adcb09a2c7
---
 mediatek/config/mt6595/autoconfig/kconfig/platform | 91 +++++++++++++++-------
 1 file changed, 62 insertions(+), 29 deletions(-)

diff --git a/mediatek/config/mt6595/autoconfig/kconfig/platform b/mediatek/config/mt6595/autoconfig/kconfig/platform
index 221deff..8f1162b 100755
--- a/mediatek/config/mt6595/autoconfig/kconfig/platform
+++ b/mediatek/config/mt6595/autoconfig/kconfig/platform
@@ -328,9 +328,6 @@ CONFIG_HMP_GLOBAL_BALANCE=y
 CONFIG_HMP_TASK_ASSIGNMENT=y
 CONFIG_HMP_DISCARD_CFS_SELECTION_RESULT=y
 # CONFIG_HMP_PACK_SMALL_TASK is not set
-# CONFIG_HMP_PACK_BUDDY_INFO is not set
-CONFIG_HMP_LAZY_BALANCE=y
-CONFIG_HMP_POWER_AWARE_CONTROLLER=y
 CONFIG_HEVTASK_INTERFACE=y
 # CONFIG_ARCH_SCALE_INVARIANT_CPU_CAPACITY is not set
 # CONFIG_HAVE_ARM_ARCH_TIMER is not set
@@ -390,7 +387,7 @@ CONFIG_USE_OF=y
 CONFIG_ATAGS=y
 # CONFIG_DEPRECATED_PARAM_STRUCT is not set
 CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE=y
-CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE_NAMES=""
+CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE_NAMES="ckt95_lw_kk"
 CONFIG_ZBOOT_ROM_TEXT=0
 CONFIG_ZBOOT_ROM_BSS=0
 # CONFIG_COMPAT_CPUINFO is not set
@@ -910,7 +907,18 @@ CONFIG_CFG80211_INTERNAL_REGDB=y
 CONFIG_CFG80211_WEXT=y
 # CONFIG_LIB80211 is not set
 # CONFIG_CFG80211_ALLOW_RECONNECT is not set
-# CONFIG_MAC80211 is not set
+CONFIG_MAC80211=y
+CONFIG_MAC80211_HAS_RC=y
+# CONFIG_MAC80211_RC_PID is not set
+CONFIG_MAC80211_RC_MINSTREL=y
+CONFIG_MAC80211_RC_MINSTREL_HT=y
+CONFIG_MAC80211_RC_DEFAULT_MINSTREL=y
+CONFIG_MAC80211_RC_DEFAULT="minstrel_ht"
+# CONFIG_MAC80211_MESH is not set
+# CONFIG_MAC80211_LEDS is not set
+# CONFIG_MAC80211_DEBUGFS is not set
+# CONFIG_MAC80211_MESSAGE_TRACING is not set
+# CONFIG_MAC80211_DEBUG_MENU is not set
 # CONFIG_WIMAX is not set
 # CONFIG_RFKILL is not set
 # CONFIG_NET_9P is not set
@@ -1221,14 +1229,39 @@ CONFIG_SLHC=y
 # CONFIG_USB_USBNET is not set
 # CONFIG_USB_IPHETH is not set
 CONFIG_WLAN=y
+# CONFIG_LIBERTAS_THINFIRM is not set
+# CONFIG_AT76C50X_USB is not set
 # CONFIG_USB_ZD1201 is not set
 # CONFIG_USB_NET_RNDIS_WLAN is not set
+# CONFIG_RTL8187 is not set
+# CONFIG_MAC80211_HWSIM is not set
 # CONFIG_WIFI_CONTROL_FUNC is not set
 # CONFIG_ATH_CARDS is not set
+# CONFIG_B43 is not set
+# CONFIG_B43LEGACY is not set
 # CONFIG_BRCMFMAC is not set
 # CONFIG_HOSTAP is not set
 # CONFIG_LIBERTAS is not set
+# CONFIG_P54_COMMON is not set
+CONFIG_RT2X00=m
+# CONFIG_RT2500USB is not set
+# CONFIG_RT73USB is not set
+CONFIG_RT2800USB=m
+CONFIG_RT2800USB_RT33XX=y
+CONFIG_RT2800USB_RT35XX=y
+CONFIG_RT2800USB_RT53XX=y
+CONFIG_RT2800USB_RT55XX=y
+CONFIG_RT2800USB_UNKNOWN=y
+CONFIG_RT2800_LIB=m
+CONFIG_RT2X00_LIB_USB=m
+CONFIG_RT2X00_LIB=m
+CONFIG_RT2X00_LIB_FIRMWARE=y
+CONFIG_RT2X00_LIB_CRYPTO=y
+CONFIG_RT2X00_LIB_LEDS=y
+CONFIG_RT2X00_DEBUG=y
+# CONFIG_RTLWIFI is not set
 # CONFIG_WL_TI is not set
+# CONFIG_ZD1211RW is not set
 # CONFIG_MWIFIEX is not set
 
 #
@@ -1688,7 +1721,8 @@ CONFIG_SND_DRIVERS=y
 CONFIG_SND_SOC=y
 # CONFIG_SND_ATMEL_SOC is not set
 CONFIG_MT_SND_SOC_V1=y
-# CONFIG_MTK_SPEAKER is not set
+CONFIG_MTK_SPEAKER=y
+# CONFIG_MTK_NXP_TFA9890 is not set
 CONFIG_SND_SOC_I2C_AND_SPI=y
 # CONFIG_SND_SOC_ALL_CODECS is not set
 CONFIG_SND_SIMPLE_CARD=y
@@ -2100,6 +2134,7 @@ CONFIG_RTC_INTF_DEV=y
 #
 CONFIG_STAGING=y
 # CONFIG_USBIP_CORE is not set
+# CONFIG_W35UND is not set
 # CONFIG_PRISM2_USB is not set
 # CONFIG_ECHO is not set
 # CONFIG_COMEDI is not set
@@ -2437,10 +2472,6 @@ CONFIG_TRACING_SUPPORT=y
 CONFIG_FTRACE=y
 CONFIG_FUNCTION_TRACER=y
 CONFIG_FUNCTION_GRAPH_TRACER=y
-CONFIG_DYNAMIC_FTRACE=y
-# CONFIG_FUNCTION_PROFILER is not set
-CONFIG_FTRACE_MCOUNT_RECORD=y
-# CONFIG_FUNCTION_DURATION_TRACER is not set
 # CONFIG_IRQSOFF_TRACER is not set
 # CONFIG_PREEMPT_TRACER is not set
 # CONFIG_SCHED_TRACER is not set
@@ -2453,6 +2484,9 @@ CONFIG_BRANCH_PROFILE_NONE=y
 # CONFIG_BLK_DEV_IO_TRACE is not set
 # CONFIG_KPROBE_EVENT is not set
 # CONFIG_PROBE_EVENTS is not set
+CONFIG_DYNAMIC_FTRACE=y
+# CONFIG_FUNCTION_PROFILER is not set
+CONFIG_FTRACE_MCOUNT_RECORD=y
 # CONFIG_FTRACE_STARTUP_TEST is not set
 # CONFIG_RING_BUFFER_BENCHMARK is not set
 # CONFIG_RING_BUFFER_STARTUP_TEST is not set
@@ -2472,6 +2506,7 @@ CONFIG_KDB_CONTINUE_CATASTROPHIC=0
 # CONFIG_TEST_KSTRTOX is not set
 # CONFIG_STRICT_DEVMEM is not set
 # CONFIG_ARM_UNWIND is not set
+CONFIG_OLD_MCOUNT=y
 # CONFIG_DEBUG_USER is not set
 CONFIG_DEBUG_RODATA=y
 # CONFIG_DEBUG_RODATA_TEST is not set
@@ -2660,7 +2695,7 @@ CONFIG_CPU_RMAP=y
 CONFIG_DQL=y
 CONFIG_NLATTR=y
 CONFIG_ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE=y
-# CONFIG_AVERAGE is not set
+CONFIG_AVERAGE=y
 # CONFIG_CORDIC is not set
 # CONFIG_DDR is not set
 # CONFIG_VIRTUALIZATION is not set
@@ -2687,6 +2722,7 @@ CONFIG_MTK_SCHED_CMP_TGS_WAKEUP=y
 CONFIG_MTK_SCHED_TRACERS=y
 CONFIG_MTK_FTRACE_DEFAULT_ENABLE=y
 # CONFIG_MTK_KERNEL_MARKER is not set
+# CONFIG_FUNCTION_DURATION_TRACER is not set
 # CONFIG_MT65XX_TRACER is not set
 
 #
@@ -2701,10 +2737,11 @@ CONFIG_MTK_MET_PLF=y
 CONFIG_MTK_MUTATION=y
 CONFIG_MTPROF_CPUTIME=y
 CONFIG_MTPROF_IRQ_DURATION=y
-# CONFIG_MTK_AEE_FEATURE is not set
-# CONFIG_MTK_AEE_AED is not set
-# CONFIG_MTK_AEE_IPANIC is not set
-# CONFIG_MTK_AEE_MRDUMP is not set
+CONFIG_MTK_AEE_FEATURE=y
+CONFIG_MTK_AEE_AED=y
+CONFIG_MTK_AEE_IPANIC=y
+CONFIG_MTK_AEE_IPANIC_TYPES=0x0fffffff
+CONFIG_MTK_AEE_MRDUMP=y
 CONFIG_MTK_AEE_POWERKEY_HANG_DETECT=y
 # CONFIG_ANDROID_VMEM is not set
 # CONFIG_CPUTIME_SUPPORT is not set
@@ -2726,7 +2763,7 @@ CONFIG_PRINTK_PROCESS_INFO=y
 CONFIG_MT_PRINTK_UART_CONSOLE=y
 CONFIG_MT_ENG_BUILD=y
 CONFIG_RAMDISK_OFFSET=0xF00000
-CONFIG_EARLY_LINUX_PORTING=y
+# CONFIG_EARLY_LINUX_PORTING is not set
 # CONFIG_MTK_MCI is not set
 # CONFIG_MTK_ETM is not set
 CONFIG_MTK_MEMCFG=y
@@ -2739,8 +2776,6 @@ CONFIG_MT_SCHED=y
 CONFIG_MT_LOAD_BALANCE_ENHANCEMENT=y
 # CONFIG_MT_LOAD_BALANCE_PROFILER is not set
 # CONFIG_MT_SCHED_NOTICE is not set
-# CONFIG_MT_SCHED_INFO is not set
-# CONFIG_MT_SCHED_DEBUG is not set
 # CONFIG_MT_SCHED_DEBUG_ONLY is not set
 CONFIG_MTK_CIRQ=y
 CONFIG_MTK_HIBERNATION=y
@@ -2749,7 +2784,6 @@ CONFIG_MTK_WQ_DEBUG=y
 CONFIG_MTK_SCHED_RQAVG_US=y
 CONFIG_MTK_SCHED_RQAVG_KS=y
 CONFIG_MTK_STORAGE_PID_LOGGER=y
-CONFIG_MTK_CPU_TOPOLOGY=y
 
 #
 # MTK Large Memory mode support
@@ -2757,6 +2791,9 @@ CONFIG_MTK_CPU_TOPOLOGY=y
 CONFIG_MTK_LM_MODE=y
 CONFIG_GATOR_DRIVER=y
 CONFIG_MTK_SYSTRACKER=y
+CONFIG_VFP_OPT=y
+CONFIG_MT_DYNAMIC_COREPATH=y
+# CONFIG_MTK_WQHD is not set
 
 #
 # MT6595 Board Support Package
@@ -2778,7 +2815,9 @@ CONFIG_MTK_FB=y
 CONFIG_MTK_FB_SUPPORT_ASSERTION_LAYER=y
 CONFIG_GPS=y
 CONFIG_MTK_GPS=y
-# CONFIG_MTK_NFC is not set
+CONFIG_MTK_NFC=y
+CONFIG_NFC_MT6605=y
+# CONFIG_NFC_MSR3110 is not set
 # CONFIG_MTK_BTIF is not set
 CONFIG_MTK_COMBO=y
 CONFIG_MTK_COMBO_CHIP_MT6620=y
@@ -2802,7 +2841,7 @@ CONFIG_MTK_I2C=y
 CONFIG_MTK_SPI=y
 CONFIG_MTK_CMMB=m
 # CONFIG_MTK_JPEG is not set
-# CONFIG_MTK_KEYPAD is not set
+CONFIG_MTK_KEYPAD=y
 # CONFIG_KEYBOARD_HID is not set
 CONFIG_MTK_LEDS=y
 CONFIG_MTK_MMC=y
@@ -2816,14 +2855,13 @@ CONFIG_MTK_MMC=y
 CONFIG_MTK_SMART_BATTERY=y
 CONFIG_MTK_PMIC=y
 CONFIG_MTK_PMIC_WRAP=y
-CONFIG_POWER_EXT=y
+# CONFIG_POWER_EXT is not set
 CONFIG_MTK_RTC=y
 CONFIG_MTK_SERIAL=y
 CONFIG_MTK_SERIAL_CONSOLE=y
 # CONFIG_MTK_SERIAL_MODEM_TEST is not set
 # CONFIG_MTK_SIM2 is not set
 # CONFIG_MTK_SOUND is not set
-# CONFIG_MTK_NXP_TFA9890 is not set
 CONFIG_MTK_TOUCHPANEL=y
 CONFIG_MTK_USB_GADGET=y
 # CONFIG_USB_MU3D_PIO_ONLY is not set
@@ -2844,8 +2882,6 @@ CONFIG_MTK_WD_KICKER=y
 # CONFIG_MT592X_SDIO_CLNT is not set
 # CONFIG_USB_MTK_ACM_TEMP is not set
 # CONFIG_USB_MTK_HDRC is not set
-CONFIG_USB_MTK_HDRC_HCD=y
-CONFIG_USB_MTK_OTG=y
 # CONFIG_USB_MTK_HDRC_GADGET is not set
 # CONFIG_USB_MTK_OTG is not set
 CONFIG_USB_MTK_DUALMODE=y
@@ -2866,12 +2902,9 @@ CONFIG_MTKPASR_MAFL=y
 CONFIG_MTKPASR_DEBUG=y
 # CONFIG_MTKPASR_NO_LASTBANK is not set
 # CONFIG_PWR_LOSS_MTK_TEST is not set
-# CONFIG_MTK_EMMC_CACHE is not set
 CONFIG_MTK_EMMC_CACHE=y
 CONFIG_PVR_RGX_MT6595=y
 # CONFIG_MTK_THERMAL_STATUS_COLLECTION is not set
 CONFIG_MD32_SUPPORT=y
 CONFIG_MTK_FAKE_CORE_COUNT=0
-CONFIG_VFP_OPT=y
-CONFIG_MT_DYNAMIC_COREPATH=y
-CONFIG_MTK_WQHD=n
+CONFIG_MTK_CPU_TOPOLOGY=y
-- 
1.9.1

